sdk/core/Azure.Core/src/Serialization/ModelSerializer.cs

```diff
@@ -173,11 +173,17 @@ private static IModelSerializable<object> GetInstance(Type returnType)
 
         private static object GetObjectInstance(Type returnType)
```

Comment:
```suggestion
        private static object GetObjectInstance([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] Type returnType)
```

You will need to annotate this `Type` as well (and any callers of this method), or else you will get trimming/AOT warnings.

BTW - have you enabled trimming/AOT analyzers for this code? See https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/prepare-libraries-for-trimming for how to get the analysis tools hooked up for a library.

cc @vitek-karas @agocke

Reply:
I currently get the following 3 warnings
```
...\ModelSerializer.cs(157): Trim analysis warning IL2067: Azure.Core.Serialization.ModelSerializer.GetInstance(Type): 'returnType' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicConstructors' in call to 'Azure.Core.Se
rialization.ModelSerializer.GetObjectInstance(Type)'. The parameter 'returnType' of method 'Azure.Core.Serialization.ModelSerializer.GetInstance(Type)' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assig
ned to.
...\ModelSerializer.cs(188): Trim analysis warning IL2067: Azure.Core.Serialization.ModelSerializer.GetObjectInstance(Type): 'type' argument does not satisfy 'DynamicallyAccessedMemberTypes.NonPublicConstructors' in call to 'System.Act
ivator.CreateInstance(Type,Boolean)'. The parameter 'returnType' of method 'Azure.Core.Serialization.ModelSerializer.GetObjectInstance(Type)' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.
...\ModelSerializer.cs(188): Trim analysis warning IL2072: Azure.Core.Serialization.ModelSerializer.GetObjectInstance(Type): 'type' argument does not satisfy 'DynamicallyAccessedMemberTypes.NonPublicConstructors' in call to 'System.Act
ivator.CreateInstance(Type,Boolean)'. The return value of method 'Azure.Core.Serialization.DefaultSubClassAttribute.DefaultSubClass.get' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.
```